homeworks/week2/Huy_Anh.java

```diff
@@ -0,0 +1,58 @@
+import java.util.*;
+
+class Main {
+    public static void main(String[] args) {
+        System.out.println(unorderedRemoveDuplicate(null)); //null
+        System.out.println(orderedRemoveDuplicate(null)); //null
+        System.out.println(unorderedRemoveDuplicate("")); //blank
+        System.out.println(orderedRemoveDuplicate("")); //blank
+        System.out.println(unorderedRemoveDuplicate("doonneeee")); //deno
+        System.out.println(orderedRemoveDuplicate("doonneeee")); //done
+        System.out.println(unorderedRemoveDuplicate("paragraphs")); //aghprs
+        System.out.println(orderedRemoveDuplicate("paragraphs")); //parghs
+        System.out.println(unorderedRemoveDuplicate("yay!!@#@'")); //!#'@ay
+        System.out.println(orderedRemoveDuplicate("yay!!@#@'")); //ya!@#'
+        System.out.println(unorderedRemoveDuplicate("hello world!")); // !dehlorw
+        System.out.println(orderedRemoveDuplicate("hello world!")); //helo wrd!
+    }
+
+    public static String orderedRemoveDuplicate(String inputString) {
+        //Create a hashtable that has all alphabet letters and numbers as keys and 0 as value
+        if (inputString == null) return inputString;
+        char[] AlphabetWithDigits = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '!', '"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',', '-', '.', '/', ':', ';', '<', '=', '>', '?', '@', '[', '\\', ']', '^', '_', '`', '{', '|', '}', '~', ' '};
```

Comment:
Managing this array is error prone. String.charAt() returns a char code of a string (integer value). You can just use that value, which make things a little easier to read.